src/main/java/core/basesyntax/dao/jdbc/CarDaoJdbcImpl.java

```diff
@@ -0,0 +1,198 @@
+package core.basesyntax.dao.jdbc;
+
+import core.basesyntax.dao.CarDao;
+import core.basesyntax.exceptions.DataProcessingException;
+import core.basesyntax.lib.Dao;
+import core.basesyntax.model.Car;
+import core.basesyntax.model.Driver;
+import core.basesyntax.model.Manufacturer;
+import core.basesyntax.util.ConnectionUtil;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Dao
+public class CarDaoJdbcImpl implements CarDao {
+    @Override
+    public Car create(Car car) {
+        String createString = "INSERT INTO cars (car_model, manufacturer_id"
+                + ") VALUES (?,?);";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(createString,
+                        Statement.RETURN_GENERATED_KEYS)) {
+            statement.setString(1, car.getModel());
+            statement.setLong(2, car.getManufacturer().getId());
+            statement.executeUpdate();
+            ResultSet resultSet = statement.getGeneratedKeys();
+            if (resultSet.next()) {
+                car.setId(resultSet.getObject(1, Long.class));
+            }
+            return car;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Unable to CREATE car in DB " + car, e);
+        }
+    }
+
+    @Override
+    public Optional<Car> get(Long id) {
+        String getCarQuery = "SELECT c.car_id, car_model AS model, c.manufacturer_id, "
+                + "manufacturer_name AS made_by, manufacturer_country AS country "
+                + "FROM cars c "
+                + "INNER JOIN manufacturers m "
+                + "ON c.manufacturer_id=m.manufacturer_id "
+                + "WHERE c.deleted=false AND c.car_id=?;";
+        Car car = null;
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement carStatement = connection.prepareStatement(getCarQuery)) {
+            carStatement.setLong(1, id);
+            ResultSet carResultSet = carStatement.executeQuery();
+            while (carResultSet.next()) {
+                car = makeCar(carResultSet);
+            }
+            List<Driver> thiCarDrivers = getCarDrivers(id);
+            car.setDrivers(thiCarDrivers);
+        } catch (SQLException e) {
```

Comment:
let's fetch and set drivers only if carResultSet.next() == true. Also let's create Car link inside try block and add return statement here too(Warning from prev. HW)

Reply:
done